### PR TITLE
Makes catwalks constructable by players.

### PR DESCRIPTION
**Pull request by CaioFrancisco**
[FEATURE] [GAME OBJECTS] [ADD TO WIKI]
## About the PR 

This PR allows players to make catwalks (and catwalk support beams) from metal rods. Catwalks and support beams can also be cut away by snipping tools. To build a catwalk (and the support beams automatically), you just have to click on a lattice with a single metal rod (making each catwalk cost roughly a metal sheet). This also changes all grilles to return only 1 steel rod when cutting, because code. (If a dev thinks this is too unreasonable, I can try to counter it with even more weird jank.)

## Why's this needed? 

The game needs it's environment to be more constructable, and because some miners or engineers may think this is a great opportunity to make stylish off-station places.

## Changelog 

```changelog
(u)Caio029
(*)Catwalks are now constructable. Click on a lattice with a steel rod and you'll start building a catwalk on it. Check minor updates for more details.
(+)While normal roles have to wait an actionbar to build the catwalk, players with engineering training don't have to wait for such actionbar.
(+)To destroy the catwalks and the support beams, click on it with wirecutters or scissors.
```

### CODE REVIEW

**Comment by Laboredih123** [comment]
damn there goes my infinitely stacked lattice barricades smh

**Comment by CaioFrancisco** [comment]
someone used lattice barricades??!??!??!???!!!!?!??!!??!

File: code/turf/floors_airless.dm [hunk 1]
```diff
@@ -30,6 +30,32 @@
 	step_material = "step_lattice"
 	step_priority = STEP_PRIORITY_MED
 
+	proc/MakeCatwalk(var/obj/item/rods)
+		if (rods)
+			rods.change_stack_amount(-1)
+
+		var/obj/grille/catwalk/catwalk = new
```
Comment (TobleroneSwordfish) [inline code/turf/floors_airless.dm:37]: materials!

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/obj/grille.dm [hunk 2]
```diff
@@ -125,6 +125,24 @@
 				if(76 to INFINITY)
 					src.icon_state += "-0"
 
+		damage_slashing(var/amount) // Completely overriding it to make it only drop 1 rod.
+			if (!isnum(amount) || amount <= 0)
+				return
+
+			if (src.ruined)
+				drop_rods(1)
```
[the remark this answers is not in the document]
Comment (CaioFrancisco) [reply code/obj/grille.dm:133]: well, it currently doesn't really work like that, currently. a grille, when it reaches 0 health after some kind of damage, sets itself to ruined and drops a rod. if it gets cut again while ruined, it drops a final rod and `qdel`s itself. it doesn't drop all rods at once. it's kinda scope creep-ey, but i'll change how it drops rods once i actually get home.

**Comment by CaioFrancisco** [comment]
Should be good to go now. It looks less janky, although the variable name could be shorter. Eh, suggest+commit in any new variable names if you want to.

**Comment by TobleroneSwordfish** [comment]
Seems like it always places the support turf facing north, leading to ugly looking patterns under the catwalk:
(top is map placed catwalk, bottom is constructed catwalk)
image
Would it be feasible to have the support turfs auto-connect alongside the catwalks?

**Comment by CaioFrancisco** [comment]
Oh, I didn't notice that. Right, I'm gonna check how auto-turfs are currently implemented (if at all) and make a separate PR for it. I'll set the PR to draft again, until I get home. If making the turfs auto-connect is not already there, then I'll close this PR and PR up auto-turfs before re-opening this PR.

**Review by TobleroneSwordfish — APPROVED** [review]

**Comment by CaioFrancisco** [comment]
Tested with the new auto catwalk supports, seems to work fine.

**Pull request merged**